v2/server.go

```diff
@@ -170,92 +172,95 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+			codecReq.WriteError(r.Context(), idx, w, http.StatusBadRequest, errRead)
+			return
+		}
+
+		// Call the registered Intercept Function
+		if s.interceptFunc != nil {
+			req := s.interceptFunc(&RequestInfo{
+				Request: r,
+				Method:  method,
+			})
+			if req != nil {
+				r = req
+			}
+		}
+
+		requestInfo := &RequestInfo{
 			Request: r,
 			Method:  method,
-		})
-		if req != nil {
-			r = req
 		}
-	}
 
-	requestInfo := &RequestInfo{
-		Request: r,
-		Method:  method,
-	}
-
-	// Call the registered Before Function
-	if s.beforeFunc != nil {
-		s.beforeFunc(requestInfo, args.Interface())
-	}
+		// Call the registered Before Function
+		if s.beforeFunc != nil {
+			s.beforeFunc(requestInfo, args.Interface())
+		}
 
-	// Prepare the reply, we need it even if validation fails
-	reply := reflect.New(methodSpec.replyType)
-	errValue := []reflect.Value{nilErrorValue}
+		// Prepare the reply, we need it even if validation fails
+		reply := reflect.New(methodSpec.replyType)
+		errValue := []reflect.Value{nilErrorValue}
 
-	// Call the registered Validator Function
-	if s.validateFunc.IsValid() {
-		errValue = s.validateFunc.Call([]reflect.Value{reflect.ValueOf(requestInfo), args})
-	}
+		// Call the registered Validator Function
+		if s.validateFunc.IsValid() {
+			errValue = s.validateFunc.Call([]reflect.Value{reflect.ValueOf(requestInfo), args})
+		}
 
-	// If still no errors after validation, call the method
-	if errValue[0].IsNil() {
-		errValue = methodSpec.method.Func.Call([]reflect.Value{
-			serviceSpec.rcvr,
-			reflect.ValueOf(r),
-			args,
-			reply,
-		})
-	}
+		// If still no errors after validation, call the method
+		if errValue[0].IsNil() {
+			errValue = methodSpec.method.Func.Call([]reflect.Value{
+				serviceSpec.rcvr,
+				reflect.ValueOf(r),
+				args,
+				reply,
+			})
+		}
 
-	// Extract the result to error if needed.
-	var errResult error
-	statusCode := http.StatusOK
-	errInter := errValue[0].Interface()
-	if errInter != nil {
-		statusCode = http.StatusBadRequest
-		errResult = errInter.(error)
-	}
+		// Extract the result to error if needed.
+		var errResult error
+		statusCode := http.StatusOK
+		errInter := errValue[0].Interface()
+		if errInter != nil {
+			statusCode = http.StatusBadRequest
+			errResult = errInter.(error)
+		}
 
-	// Prevents Internet Explorer from MIME-sniffing a response away
-	// from the declared content-type
-	w.Header().Set("x-content-type-options", "nosniff")
+		// Prevents Internet Explorer from MIME-sniffing a response away
+		// from the declared content-type
+		w.Header().Set("x-content-type-options", "nosniff")
 
-	// Encode the response.
-	if errResult == nil {
-		codecReq.WriteResponse(w, reply.Interface())
-	} else {
-		codecReq.WriteError(r.Context(), w, statusCode, errResult)
-	}
+		// Encode the response.
+		if errResult == nil {
+			codecReq.WriteResponse(idx, w, reply.Interface())
+		} else {
+			codecReq.WriteError(r.Context(), idx, w, statusCode, errResult)
```

Comment:
```suggestion
			codecReq.WriteError(r.Context(), idx, w, statusCode, errResult)
			return
```

We should return correct?

Reply:
No, we need to process all the requests and return all the response. The requests may not be related.